myhoard/backup_stream.py

```diff
@@ -69,6 +69,17 @@ class PromotionInfo(TypedDict):
     start_index: int
 
 
+class BaseBackupFailureReason(str, enum.Enum):
+    network_error = "network_error"
+    block_mismatch_error = "block_mismatch_error"
+    xtrabackup_error = "xtrabackup_error"
+
+
+class BaseBackupAttempt(TypedDict):
+    failed_at: float
+    reason: BaseBackupFailureReason
```

Comment:
**comment** I might remove this, maybe keeping the reason of the failures might help the BackupStream to know what to do on each particular case, but still not sure about it